pkg/relayer/proxy/sync.go

```diff
@@ -227,3 +234,107 @@ func (server *relayMinerHTTPServer) sendRelayResponse(
 	_, err = writer.Write(relayResponseBz)
 	return err
 }
+
+// forwardPayload represents the HTTP request body format to forward a request
+// to the supplier.
+type forwardPayload struct {
```

Comment:
can we move this into `pkg/relayer/proxy/types.go` along with the helpers below?